src/protocol/sort/multi_bit_permutation.rs

```diff
@@ -0,0 +1,175 @@
+use crate::{
+    error::Error,
+    ff::Field,
+    protocol::{
+        boolean::multiply_all_shares,
+        context::Context,
+        sort::MultiBitPermutationStep::{MultiplyAcrossBits, Sop},
+        RecordId,
+    },
+    secret_sharing::SecretSharing,
+};
+
+use futures::future::try_join_all;
+
+/// This is an implementation of `GenMultiBitSort` (Algorithm 11) described in:
+/// "An Efficient Secure Three-Party Sorting Protocol with an Honest Majority"
+/// by K. Chida, K. Hamada, D. Ikarashi, R. Kikuchi, N. Kiribuchi, and B. Pinkas
+/// <https://eprint.iacr.org/2019/695.pdf>.
+///
+/// Protocol to compute a secret sharing of a permutation, after sorting on multiple bits `num_multi_bits`.
+/// At a high level, the protocol works as follows:
+/// 1. Start with a vector of list of `L*n` secret shares `[[x1_1 ... x1_n], .. , [xL_1 ... xL_n]]` where each is a secret sharing of either zero or one.
+/// 2. For j in 0 to 2 pow `num_multi_bits`
+///    i. Get binary representation of j (Bk)
+///    ii. For i in `num_multi_bits`
+///      a. Locally compute `mult_inputs` as (Bk * k + (1-Bk)(1- k))
```

Comment:
Reading this, I lose you here. Not sure what "k" is, or how it relates to Bk.

Reply:
Tried to make it better.